# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/generic/DocAccessible-inl.h
# Commit: 1e75ce1fef67
# Full Hash: 1e75ce1fef67f3139ac4f05993cad693c82c111c
# Author: Eitan Isaacson <eitan@monotonous.org>
# Date: 2020-02-17 09:50:03
# Regressor Bug: 1611060
# File Overlap Count: 1
# Description:
#   Bug 1611060 - Dispatch a11y scroll events for all DOM scrolls. r=Jamie
#   
#   Currently, only documents dispatch scroll events when in reality any
#   element can have scrollable overflows.
#   
# ==============================================================================

diff -r 20b59309d020 -r 1e75ce1fef67 accessible/generic/DocAccessible-inl.h
--- a/accessible/generic/DocAccessible-inl.h	Mon Feb 17 05:37:24 2020 +0200
+++ b/accessible/generic/DocAccessible-inl.h	Mon Feb 17 04:09:34 2020 +0000
@@ -77,27 +77,6 @@
     mNotificationController->ScheduleTextUpdate(aTextNode);
 }
 
-inline void DocAccessible::AddScrollListener() {
-  // Delay scroll initializing until the document has a root frame.
-  if (!mPresShell->GetRootFrame()) return;
-
-  mDocFlags |= eScrollInitialized;
-  nsIScrollableFrame* sf = mPresShell->GetRootScrollFrameAsScrollable();
-  if (sf) {
-    sf->AddScrollPositionListener(this);
-
-#ifdef A11Y_LOG
-    if (logging::IsEnabled(logging::eDocCreate))
-      logging::Text("add scroll listener");
-#endif
-  }
-}
-
-inline void DocAccessible::RemoveScrollListener() {
-  nsIScrollableFrame* sf = mPresShell->GetRootScrollFrameAsScrollable();
-  if (sf) sf->RemoveScrollPositionListener(this);
-}
-
 inline void DocAccessible::NotifyOfLoad(uint32_t aLoadEventType) {
   mLoadState |= eDOMLoaded;
   mLoadEventType = aLoadEventType;